### PR TITLE
Remove tests from sql_integration that were ported to sqllogictest 

### LINKED ISSUE #4498: Remove tests from `sql_integration` that were ported to `sqllogictest`
https://github.com/apache/arrow-datafusion/pull/4395 moved several initial tests from sql_integration into sqllogictests

Specifically, tests that were moved include:
* https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/sql/aggregates.rs --> https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/sqllogictests/test_files/aggregate.slt
* https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/sql/arrow_typeof.rs --> https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/sqllogictests/test_files/arrow_typeof.slt

The goal of this ticket is to:
1. Review that all tests were moved (and none were added subsequently, for example)
2. Port any remaining gaps
3. Remove the `.rs` files

### PR DESCRIPTION
**Pull request by matthewwillian**
# Which issue does this PR close?

Closes #4498

# Rationale for this change

# What changes are included in this PR?

- Removes arrow_typeof.rs
- Deletes a number of tests in aggegates.rs
- Uncomments tests in aggregate.slt that worked without any substantial changes

# Are these changes tested?

Unit-tests pass

# Are there any user-facing changes?

No

### CODE REVIEW

**Review by jackwener — COMMENTED** [review]
The newly added sqllogicaltest does not cover all the removed tests.

**Comment by alamb** [comment]
Thanks @matthewwillian  and @jackwener  -- I plan to review this PR tomorrow as well.

**Comment by matthewwillian** [comment]
Thanks @jackwener. One of the aggregates tests and all of the arrow_typeof tests are duplicates of tests that already exist.

File: datafusion/core/tests/sql/aggregates.rs [hunk 1]
```diff
@@ -20,24 +20,6 @@ use datafusion::scalar::ScalarValue;
 use datafusion::test_util::scan_empty;
 use datafusion_common::cast::as_float64_array;
 
-#[tokio::test]
-async fn csv_query_avg_multi_batch() -> Result<()> {
```
Comment (alamb) [inline datafusion/core/tests/sql/aggregates.rs:24]: Here is the corresponding test in .slt

https://github.com/apache/arrow-datafusion/blob/dcd52ee3d87c4dd9e2c176165e9e20644f66988b/datafusion/core/tests/sqllogictests/test_files/aggregate.slt#L21-L25

👍

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 2]
```diff
@@ -846,26 +846,23 @@ SELECT count(1 + 1)
 ----
 1
 
-# FIX: "CSV Writer does not support List(Field { name: \"item\", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: None }) data type")
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:849]: 👍

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 3]
```diff
@@ -976,6 +973,7 @@ select max(c1) from d_table
 ----
 110.009
 
+# FIX: doesn't check datatype
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:976]: This is tracked by https://github.com/apache/arrow-datafusion/issues/4499

File: datafusion/core/tests/sql/arrow_typeof.rs [hunk 4]
```diff
@@ -1,139 +0,0 @@
...
-    let sql = "SELECT arrow_typeof(null)";
-    let actual = execute(&ctx, sql).await;
-    let expected = "Null";
-    assert_eq!(expected, &actual[0][0]);
-
-    Ok(())
-}
-
-#[tokio::test]
-async fn arrow_typeof_boolean() -> Result<()> {
```
Comment (alamb) [inline datafusion/core/tests/sql/arrow_typeof.rs:32]: I double checked -- here is the corresponding .slt file: 

https://github.com/apache/arrow-datafusion/blob/dcd52ee3d87c4dd9e2c176165e9e20644f66988b/datafusion/core/tests/sqllogictests/test_files/arrow_typeof.slt#L26

**Review by alamb — APPROVED** [review]
Thank you @matthewwillian  -- this looks great to me. I double checked the tests and 👍 

Deleting code is a wonderful first contribution 🎉

**Comment by alamb** [comment]
It appears there are some CI checks failing on this PR

**Review by jackwener — APPROVED** [review]
> Thanks @jackwener. One of the aggregates tests and all of the arrow_typeof tests are duplicates of tests that already exist.

Looks great to me, cleanup the useless code is great to me. 

I have checked the tests. Thanks @matthewwillian.

**Pull request merged**